models/stg_apple_store__sales_subscription_events.sql

```diff
@@ -30,21 +30,38 @@ final as (
     select
         cast(source_relation as {{ dbt.type_string() }}) as source_relation, 
         cast(event_date as date) as date_day,
-        cast(account_number as {{ dbt.type_bigint() }}) as account_id,
+        cast(_fivetran_id as {{ dbt.type_string() }}) as _fivetran_id,
+        cast(vendor_number as {{ dbt.type_int() }}) as vendor_number,
+        cast(app_apple_id as {{ dbt.type_int() }}) as app_apple_id,
         cast(app_name as {{ dbt.type_string() }}) as app_name,
-        cast(subscription_name as {{ dbt.type_string() }}) as subscription_name,
         cast(event as {{ dbt.type_string() }}) as event,
+        cast(subscription_name as {{ dbt.type_string() }}) as subscription_name,
+        cast(subscription_apple_id as {{ dbt.type_int() }}) as subscription_apple_id,
+        cast(subscription_group_id as {{ dbt.type_int() }}) as subscription_group_id,
+        cast(standard_subscription_duration as {{ dbt.type_string() }}) as standard_subscription_duration,
+        cast(subscription_offer_type as {{ dbt.type_string() }}) as subscription_offer_type,
+        cast(subscription_offer_duration as {{ dbt.type_string() }}) as subscription_offer_duration,
+        cast(marketing_opt_in as {{ dbt.type_string() }}) as marketing_opt_in,
+        cast(marketing_opt_in_duration as {{ dbt.type_string() }}) as marketing_opt_in_duration,
+        cast(preserved_pricing as {{ dbt.type_string() }}) as preserved_pricing,
+        cast(proceeds_reason as {{ dbt.type_string() }}) as proceeds_reason,
+        cast(promotional_offer_name as {{ dbt.type_string() }}) as promotional_offer_name,
+        cast(promotional_offer_id as {{ dbt.type_string() }}) as promotional_offer_id,
+        cast(consecutive_paid_periods as {{ dbt.type_int() }}) as consecutive_paid_periods,
+        cast(original_start_date as date) as original_start_date,
+        cast(device as {{ dbt.type_string() }}) as device,
+        cast(client as {{ dbt.type_string() }}) as client,
+        cast(state as {{ dbt.type_string() }}) as state,
```

Comment:
Can you confirm the `state` and `quantity` case statements are no longer necessary in this schema?

Reply:
I assume you meant the `device` case statement rather than `quantity`; for that one, I didn't see any values needing cleanup. 

For `state`, actually now I noticed `' ' ` present in the data so thanks for calling attention to that-- I will add that case logic back for `state`.